addons/mail/models/res_users.py

```diff
@@ -273,7 +273,9 @@ def _init_messaging(self):
 
     @api.model
     def systray_get_activities(self):
-        activities = self.env["mail.activity"].search([("user_id", "=", self.env.uid)])
+        search_limit = int(self.env['ir.config_parameter'].sudo().get_param('mail.activity.systray.limit', 1000))
+        activities = self.env["mail.activity"].search(
+            [("user_id", "=", self.env.uid)], order='id desc', limit=search_limit)
```

Comment:
Does the order actually matter?

Reply:
As we have decided to limit the number of activities retrieved, the idea was to display only the last one if there were too many. What do you think ?